TheJumpCloud/jcapi-powershell
# Application custom config

## Issues
* [CUT-3922](https://jumpcloud.atlassian.net/browse/CUT-3922) - Add additional application (sso) properties to 

## What does this solve?

The default models defined in our API specification are missing several properties in the Application model, This release manually re-defines the `config` model for the Application model.

## Is there anything particularly tricky?

I've added a new "find and replace" type of way to redefine the generated code. Instead of using regex, to find and replace many things, you just need to replace sections of the API model. Within the `apiTransform.ps1` file I've added a `OverrideDefinitions` list. Within that list, one can denote a part of the SDK swagger spec to replace entirely. All you would need to do that would be to define a `.json` file with the same name in the `/Custom` directory. 

## How should this be tested?

Using the current version of the V2 SDK (0.0.43) running the command: `remove-JCSystemGroupMember -SystemID 62ace7fbac30e538a6ba0ec8 -Groupid 604938c5232e1179d1e7bbdb` should return multiple status messages
![Screenshot 2024-03-28 at 9 31 12 AM](https://github.com/TheJumpCloud/jcapi-powershell/assets/54448601/a2bac51b-644c-4f82-8cf2-13c984c01b69)

Using this version of the V2 SDK (0.0.44) running the command: `remove-JCSystemGroupMember -SystemID 62ace7fbac30e538a6ba0ec8 -Groupid 604938c5232e1179d1e7bbdb` should return the single status message as expected
![Screenshot 2024-03-28 at 9 30 45 AM](https://github.com/TheJumpCloud/jcapi-powershell/assets/54448601/b6f6617d-b348-41e7-aaaf-e70c88090246)

503 error should still catch and recover 😎
![Screenshot 2024-03-28 at 9 32 43 AM](https://github.com/TheJumpCloud/jcapi-powershell/assets/54448601/4a4dda7e-82a3-4d4c-86a1-bf43617374a9)


## Screenshots

[CUT-3922]: https://jumpcloud.atlassian.net/browse/CUT-3922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
